.devcontainer/Dockerfile

```diff
@@ -13,4 +14,12 @@ RUN apt update -y && apt upgrade -y \
     libgtest-dev \
     # Dependency for intel-mkl
     libomp-dev \
-  && rm -rf /var/lib/apt/lists/*
+    # Dependency for clangd-tidy
+    python3 \
+    python3-pip \
+    python3-venv \
+  && rm -rf /var/lib/apt/lists/* 
+  
+RUN python3 -m venv /.venv \
```

Comment:
RUN commands should be combined.

Reply:
Done. But in general, I don't think this would have made too much of a difference.

Reply:
It creates excess docker images in the runner cache.